### PR TITLE
Multiple series images

### PR DESCRIPTION
**Pull request by kookster**
- [x] Update models to add default image and has_many for series images
- [x] Update routes for series images to be resource(s)
- [x] Update representers
- [x] Add "use" or "purpose" attribute to model and filter for controller
- [x] Determine default image using purpose as well
- [x] Send the correct images to feeder based on `purpose`
- [x] Define common image purposes
- [x] Add filter by purpose to image controllers
- [x] Set both profile and thumbnail images on podcast

### CODE REVIEW

**Comment by kookster** [comment]
I used `purpose` instead of `use` because `use` is a SQL reserved word and caused some headaches.

File: app/controllers/api/series_images_controller.rb [hunk 1]
```diff
@@ -1,19 +1,11 @@
 # encoding: utf-8
 
 class Api::SeriesImagesController < Api::BaseController
-  include Announce::Publisher
```
Comment (cavis) [inline app/controllers/api/series_images_controller.rb]: Really simplifies this logic.

File: app/models/image.rb [hunk 2]
```diff
@@ -12,6 +12,18 @@ class Image < BaseModel
   FAILED = 'failed'.freeze
   COMPLETE = 'complete'.freeze
 
+  PROFILE = 'profile'.freeze
+  THUMBNAIL = 'thumbnail'.freeze
+  PURPOSES = [PROFILE, THUMBNAIL].freeze
+
+  def self.profile
```
Comment (cavis) [inline app/models/image.rb]: So you're adding the `purpose` field to every image type, then?

Also, is profile-vs-thumbnail a generic enough relation to warrant having them in every image sub-type?  (Mostly playing devil's advocate here)

File: app/representers/api/series_representer.rb [hunk 3]
```diff
@@ -26,18 +26,19 @@ class Api::SeriesRepresenter < Api::BaseRepresenter
...
+  link :images do
     {
-      href: api_series_series_image_path(represented),
-      title: 'Create an image'
-    } if represented.id && !represented.image
+      href: api_series_series_images_path(represented),
+      count: represented.images.count
+    } if represented.id
   end
+  embed :images, paged: true, item_class: SeriesImage, item_decorator: Api::ImageRepresenter
```
Comment (cavis) [inline app/representers/api/series_representer.rb]: Is it necessary to embed all series-images (usually 2) in this response?  I think the pattern for Story is to embed the default (the 1st by position), and just link/count the others.

File: app/models/image.rb (hunk 2 above)
Comment (kookster) [inline app/models/image.rb]: yes, I think so, I thought about it.

File: app/representers/api/series_representer.rb (hunk 3 above)
Comment (kookster) [inline app/representers/api/series_representer.rb]: Yeah, I updated this to embed them all b/c it is as common to the podcast as the audio files are to a story, i.e. I am going to end up in feeder having to retrieve them both every time there is an update. They are not all in the min representer, so when included in a story or a list they don't appear.

**Pull request merged**